prov-api/helpers/generate_report_helpers.py

```diff
@@ -0,0 +1,702 @@
+    """
+
+    try:
+        os.remove(file_path)
+
+    except FileNotFoundError as e: 
+        raise HTTPException(status_code=500, detail=f"Unable to delete the file with path {file_path} - {str(e)}")
+    
+    except Exception as e: 
+        raise HTTPException(status_code=500, detail=f"Something has gone wrong in deleting the file {str(e)}")
+
+def get_model_runs_from_study(study_node_id:str, config:Config, depth:int = 1) -> List[Node]: 
+    """
+    Fetches all model runs associated with the respective study.
+
+    Parameters
+    ----------
+    study_node_id : str
+        The ID of the study node.
+    config : Config
+        A config object containing information about the different endpoints of the system.
+    depth : int, optional
+        The depth of the downstream query to traverse to, by default 1
+
+    Returns
+    -------
+    List[Node]
+        A list of all model run nodes associated with the study.
+
+    Raises
+    ------
+    HTTPException
+        If no model run nodes are found in the downstream query.
+    HTTPException
+        If any error occurs during fetching.
+    """
+
+    try:
+        # Query downstream to get all linked model runs at depth 1. 
+        # This may contain more than one model run.
+        downstream_model_run_response = downstream_query(starting_id=study_node_id, depth=depth, config=config)
+        assert downstream_model_run_response.get('nodes'), f"The study with id {study_node_id} does not have any associated model runs."
+
+        model_run_nodes: List[Node] = parse_nodes(downstream_model_run_response.get('nodes', []))
+        return model_run_nodes
+    
+    except AssertionError as e:
+        raise HTTPException(status_code=404, detail=f"{str(e)}")
+    
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=f"Error fetching downstream nodes from the provided study with id {study_node_id} - error: {str(e)}")
+
+async def populate_model_run_report(
+    user: ProtectedRole,
+    node_id: str,
+    upstream_depth: int,
+    config: Config,
+    report_node_collection: ReportNodeCollection
+) -> None:
+    """
+    A helper function wrapper around the "generate-report" helper and prepares the required parameters.
+
+    Parameters
+    ----------
+    user : ProtectedRole
+        The role of the user making the request, used for making requests to Registry API.
+    node_id : str
+        The ID of the model run node.
+    upstream_depth : int
+        The depth of the upstream query to traverse upto
+    config : Config
+        A config object containing information about the different endpoints of the system.
+    report_node_collection : ReportNodeCollection
+        A collection to store and manage parsed nodes for report generation.
+    """
+
+    await generate_report(
+        user=user,
+        starting_id=node_id,
+        upstream_depth=upstream_depth,
+        config=config,
+        report_node_collection=report_node_collection
+    )
```

Comment:
The reason for not returning a value is because I have passed the instantiated dataclass as the `report_node_collection`, and all modifications such as addition of nodes are then directly done on the dataclass instance, so no need to return. 

The reason for having a "shared" data class or passing the instantiated data class was so it's easier to manage when a study is linked to multiple model runs, but the trade-off is then for single model run traversals it follows the shared data class pattern.